pyk/src/pyk/rewrite.py

```diff
@@ -46,6 +55,54 @@ def replace(term):
 
         return bottom_up(replace, term)
 
+    def unapply(self, term: KInner) -> KInner:
+        new_term = term
+        for var_name in self:
+            lhs = self[var_name]
+            if type(lhs) is not KToken:
+                rhs = KVariable(var_name)
+                new_term = replace_anywhere((lhs, rhs), new_term)
```

Comment:
We don't need this `KToken` check anymore here, because `extract_subst` does not consider substitutions for `KToken` anymore.

Reply:
Done.